### ISSUE TITLE
Catalog signing verification is not working in v6.0.0-beta.1 release of PowerShellCore

### ISSUE THREAD

**Issue by bmanikm** [PowerShell/PowerShellGetv2#125]
The following features of PowerShellGet are broken in v6.0.0-beta.1 release of PowerShellCore due to https://github.com/PowerShell/PowerShell/pull/3556 .
- Module catalog signing verification on Windows.
- Proxy
- Telemetry 

Failed tests can be found at https://ci.appveyor.com/project/PowerShell/powershellget-0lib3/branch/development/job/j9ff508mkf21b3ym/tests .

**Comment by daxian-dbw** [comment]
Is there a simple repro for the regression? (like using `Get-AuthenticodeSignature`)

**Comment by bmanikm** [comment]
## Simple repro steps
```powershell
$requiredAssembly = @([System.Management.Automation.PSCmdlet].Assembly.FullName)
$source = @" 
namespace Microsoft.PowerShell.Commands.PowerShellGet 
{ 
    public static class Telemetry  
    { 
        public static void TraceMessageArtifactsNotFound(string[] artifactsNotFound, string operationName) 
        { 
            Microsoft.PowerShell.Telemetry.Internal.TelemetryAPI.TraceMessage(operationName, new { ArtifactsNotFound = artifactsNotFound });
        }         
        
        public static void TraceMessageNonPSGalleryRegistration(string sourceLocationType, string sourceLocationHash, string installationPolicy, string packageManagementProvider, string publishLocationHash, string scriptSourceLocationHash, string scriptPublishLocationHash, string operationName) 
        { 
            Microsoft.PowerShell.Telemetry.Internal.TelemetryAPI.TraceMessage(operationName, new { SourceLocationType = sourceLocationType, SourceLocationHash = sourceLocationHash, InstallationPolicy = installationPolicy, PackageManagementProvider = packageManagementProvider, PublishLocationHash = publishLocationHash, ScriptSourceLocationHash = scriptSourceLocationHash, ScriptPublishLocationHash = scriptPublishLocationHash });
        }         
    }
} 
"@ 

Add-Type -TypeDefinition $source -Language CSharp -Verbose -PassThru -ReferencedAssemblies $requiredAssembly 
```


```powershell
C:\WINDOWS\system32>"c:\Program Files\PowerShell\6.0.0-beta.1\powershell.exe"
PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

PS C:\WINDOWS\system32> $requiredAssembly = @([System.Management.Automation.PSCmdlet].Assembly.FullName)
PS C:\WINDOWS\system32> $source = @"
>> namespace Microsoft.PowerShell.Commands.PowerShellGet
>> {
>>     public static class Telemetry
>>     {
>>         public static void TraceMessageArtifactsNotFound(string[] artifactsNotFound, string operationName)
>>         {
>>             Microsoft.PowerShell.Telemetry.Internal.TelemetryAPI.TraceMessage(operationName, new { ArtifactsNotFound = artifactsNotFound });
>>         }
>>
>>         public static void TraceMessageNonPSGalleryRegistration(string sourceLocationType, string sourceLocationHash, string installationPolicy, string packageManagementProvider, string publishL
ocationHash, string scriptSourceLocationHash, string scriptPublishLocationHash, string operationName)
>>         {
>>             Microsoft.PowerShell.Telemetry.Internal.TelemetryAPI.TraceMessage(operationName, new { SourceLocationType = sourceLocationType, SourceLocationHash = sourceLocationHash, InstallationP
olicy = installationPolicy, PackageManagementProvider = packageManagementProvider, PublishLocationHash = publishLocationHash, ScriptSourceLocationHash = scriptSourceLocationHash, ScriptPublishLocat
ionHash = scriptPublishLocationHash });
>>         }
>>     }
>> }
>> "@
PS C:\WINDOWS\system32>
PS C:\WINDOWS\system32> Add-Type -TypeDefinition $source -Language CSharp -Verbose -PassThru -ReferencedAssemblies $requiredAssembly
Add-Type : (1) : Missing compiler required member 'System.Collections.Generic.EqualityComparer`1.Equals'
(1) : >>> namespace Microsoft.PowerShell.Commands.PowerShellGet
(2) : {
At line:1 char:1
+ Add-Type -TypeDefinition $source -Language CSharp -Verbose -PassThru  ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (Microsoft.Power...peCompilerError:AddTypeCompilerError) [Add-Type], Exception
    + FullyQualifiedErrorId : SOURCE_CODE_ERROR,Microsoft.PowerShell.Commands.AddTypeCommand

Add-Type : (1) : Missing compiler required member 'System.Collections.Generic.EqualityComparer`1.GetHashCode'
(1) : >>> namespace Microsoft.PowerShell.Commands.PowerShellGet
(2) : {
At line:1 char:1
+ Add-Type -TypeDefinition $source -Language CSharp -Verbose -PassThru  ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (Microsoft.Power...peCompilerError:AddTypeCompilerError) [Add-Type], Exception
    + FullyQualifiedErrorId : SOURCE_CODE_ERROR,Microsoft.PowerShell.Commands.AddTypeCommand

Add-Type : (1) : Missing compiler required member 'System.Collections.Generic.EqualityComparer`1.get_Default'
(1) : >>> namespace Microsoft.PowerShell.Commands.PowerShellGet
(2) : {
At line:1 char:1
+ Add-Type -TypeDefinition $source -Language CSharp -Verbose -PassThru  ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (Microsoft.Power...peCompilerError:AddTypeCompilerError) [Add-Type], Exception
    + FullyQualifiedErrorId : SOURCE_CODE_ERROR,Microsoft.PowerShell.Commands.AddTypeCommand

Add-Type : Cannot add type. Compilation errors occurred.
At line:1 char:1
+ Add-Type -TypeDefinition $source -Language CSharp -Verbose -PassThru  ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Add-Type], InvalidOperationException
    + FullyQualifiedErrorId : COMPILER_ERRORS,Microsoft.PowerShell.Commands.AddTypeCommand

PS C:\WINDOWS\system32>
PS C:\WINDOWS\system32> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.0.0-beta
PSEdition                      Core
BuildVersion                   3.0.0.0
CLRVersion
GitCommitId                    v6.0.0-beta.1
OS                             Microsoft Windows 10.0.14393
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0


PS C:\WINDOWS\system32>
```

## Required resolution as per PowerShell/PowerShell#3556 .
As per PowerShell/PowerShell#3556, I think, it is not required to specify -ReferencedAssemblies for .Net assemblies. 

### Resolution
```powershell
$source = @" 
namespace Microsoft.PowerShell.Commands.PowerShellGet 
{ 
    public static class Telemetry  
    { 
        public static void TraceMessageArtifactsNotFound(string[] artifactsNotFound, string operationName) 
        { 
            Microsoft.PowerShell.Telemetry.Internal.TelemetryAPI.TraceMessage(operationName, new { ArtifactsNotFound = artifactsNotFound });
        }         
        
        public static void TraceMessageNonPSGalleryRegistration(string sourceLocationType, string sourceLocationHash, string installationPolicy, string packageManagementProvider, string publishLocationHash, string scriptSourceLocationHash, string scriptPublishLocationHash, string operationName) 
        { 
            Microsoft.PowerShell.Telemetry.Internal.TelemetryAPI.TraceMessage(operationName, new { SourceLocationType = sourceLocationType, SourceLocationHash = sourceLocationHash, InstallationPolicy = installationPolicy, PackageManagementProvider = packageManagementProvider, PublishLocationHash = publishLocationHash, ScriptSourceLocationHash = scriptSourceLocationHash, ScriptPublishLocationHash = scriptPublishLocationHash });
        }         
    }
} 
"@ 

Add-Type -TypeDefinition $source -Language CSharp -Verbose -PassThru
```


```powershell
PS C:\WINDOWS\system32> $source = @"
>> namespace Microsoft.PowerShell.Commands.PowerShellGet
>> {
>>     public static class Telemetry
>>     {
>>         public static void TraceMessageArtifactsNotFound(string[] artifactsNotFound, string operationName)
>>         {
>>             Microsoft.PowerShell.Telemetry.Internal.TelemetryAPI.TraceMessage(operationName, new { ArtifactsNotFound = artifactsNotFound });
>>         }
>>
>>         public static void TraceMessageNonPSGalleryRegistration(string sourceLocationType, string sourceLocationHash, string installationPolicy, string packageManagementProvider, string publishL
ocationHash, string scriptSourceLocationHash, string scriptPublishLocationHash, string operationName)
>>         {
>>             Microsoft.PowerShell.Telemetry.Internal.TelemetryAPI.TraceMessage(operationName, new { SourceLocationType = sourceLocationType, SourceLocationHash = sourceLocationHash, InstallationP
olicy = installationPolicy, PackageManagementProvider = packageManagementProvider, PublishLocationHash = publishLocationHash, ScriptSourceLocationHash = scriptSourceLocationHash, ScriptPublishLocat
ionHash = scriptPublishLocationHash });
>>         }
>>     }
>> }
>> "@
PS C:\WINDOWS\system32>
PS C:\WINDOWS\system32> Add-Type -TypeDefinition $source -Language CSharp -Verbose -PassThru

IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
False    False    <>f__AnonymousType0`1                    System.Object
False    False    <>f__AnonymousType1`7                    System.Object
True     False    Telemetry                                System.Object
```

**Comment by bmanikm** [comment]
This issue got resolved in PowerShellGet with PR #124.

**Comment by daxian-dbw** [comment]
Ah, so the root cause is `Add-Type`.
It's completely expected for `Add-Type` to break when we move to a new .NET Core version **if `-ReferencedAssemblies` is explicitly specified**, because .NET types and assemblies get shuffled consistently. That's why it was fixed in PowerShell/PowerShell#3556 to no longer require users to specify `-ReferencedAssemblies` in PowerShell Core.
Glad that the issue is understood and resolved.

/cc @joeyaiello
